mbed/mbed.py

```diff
@@ -2667,26 +2728,37 @@ def test_(toolchain=None, target=None, compile_list=False, run_list=False, compi
                   + list(chain.from_iterable(zip(repeat('--source'), source)))
                   + ['--build', build_path]
                   + ['--test-spec', test_spec]
-                  + (['-n', tests_by_name] if tests_by_name else [])
+                  + (['--build-data', build_data] if build_data else [])
+                  + (['-n', tests_by_name_temp] if tests_by_name_temp else [])
                   + (['-v'] if verbose else [])
                   + (['--app-config', app_config] if app_config else [])
                   + (['--test-config', test_config] if test_config else [])
+                  + (['--icetea'] if icetea_supported and icetea else [])
+                  + (['--greentea'] if icetea_supported and greentea else [])
                   + args,
                   env=env)
 
-        if run_list:
-            popen(['mbedgt', '--test-spec', test_spec, '--list']
-                  + (['-n', tests_by_name] if tests_by_name else [])
-                  + (['-V'] if verbose else [])
-                  + args,
-                  env=env)
+        # Greentea tests
+        if greentea:
+            greentea_command = ['mbedgt', '--test-spec', test_spec] \
```

Comment:
Looks like the `if run_list` check has been dropped for greentea here. Could this be added back since it is present for icetea below?

Reply:
I don't get the idea.. can you explain a bit more?